### PR TITLE
feat: add translations process docs

### PR DESCRIPTION
**Pull request by Carlos-Muniz**
This PR adds translations-process.rst which contains a brief summary of the newly proposed translation process.

closes https://github.com/openedx/docs.openedx.org/issues/132

### CODE REVIEW

File: source/translators/concepts/translation-process.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
+Translations Process
+####################
+
+High Level
```
Comment (feanil) [inline source/translators/concepts/translation-process.rst:4]:
```suggestion
Overview
```
Overview is a much more common heading for this section.

File: source/translators/concepts/translation-process.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+**********
+
+Your code needs to annotate strings so tooling can know that you want those strings
```
Comment (feanil) [inline source/translators/concepts/translation-process.rst:7]:
```suggestion
The Open edX Platform supports running in multiple languages.  This page explains
the process by which we accomplish this.  To begin with, your code needs to annotate 
strings so tooling can know that you want those strings
```

I realized a bit of orienting information would also be useful.

File: source/translators/concepts/translation-process.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+   # over multiple lines because it would be too long otherwise.
+   message = _("Hello world")
+
+Our tooling will extract the following strings:
+
+* ``Welcome!``
+
+* ``Hello world``
+
+For our django backed services, we use `django's builtin translation tooling`_ to extract
```
Comment (feanil) [inline source/translators/concepts/translation-process.rst:44]:
```suggestion
A similar process to the one noted above occurs in our Javascript Fronteend code as well.
For our django backed services, we use `django's builtin translation tooling`_ to extract
```

File: source/translators/concepts/translation-process.rst [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+strings for translation. In our react based frontends, we use react-intl_ to extract
+strings for translation.
+
+To learn more about how to annotate strings for translation, see the annotation reference
+(link tbd).
```
Comment (feanil) [inline source/translators/concepts/translation-process.rst:49]: We wrote https://github.com/openedx/docs.openedx.org/issues/211 to track this work, maybe put a link to that in a comment here.

**Review by feanil — APPROVED** [review]
Some suggestions but nothing blocking.

File: source/translators/concepts/translation-process.rst (hunk 4 above)
Comment (Carlos-Muniz) [reply source/translators/concepts/translation-process.rst:52]: Added a comment with the link to the issue.

**Pull request merged**